### PR TITLE
Converted AdminRootController to PSR-2

### PR DESCRIPTION
**Pull request by assertchris**
This mainly converts `AdminRootController` to PSR-2, also limiting line length to 100 and adding a few type hints.

I recommend reviewing it at [this link](https://github.com/silverstripe/silverstripe-framework/pull/5190/files?w=1).

This is a test-case for [a wider discussion](https://groups.google.com/forum/#!topic/silverstripe-dev/IoMtKlju3Rc).

### CODE REVIEW

File: admin/code/AdminRootController.php [hunk 1]
```diff
@@ -4,96 +4,117 @@
...
+    /**
+     * Holds an array of url_pattern => controller k/v pairs, the same as Director::rules. However
+     * this is built dynamically from introspecting on all the classes that derive from LeftAndMain.
+     *
+     * Don't access this directly - always access via the rules() accessor below, which will build
+     * this array the first time it's accessed.
+     *
+     * @var array
+     */
+    private static $_rules = null;
```
Comment (kinglozzer) [inline admin/code/AdminRootController.php]: > Property names SHOULD NOT be prefixed with a single underscore to indicate protected or private visibility.
http://www.php-fig.org/psr/psr-2/#4-2-properties

Should be fine to rename this to `$rules` as it’s private

File: admin/code/AdminRootController.php [hunk 2]
```diff
@@ -4,96 +4,117 @@
...
+     * @param string $controllerClass
+     */
+    protected static function add_rule_for_controller($controllerClass)
+    {
+        $urlSegment = Config::inst()->get($controllerClass, 'url_segment', Config::FIRST_SET);
+        $urlRule = Config::inst()->get($controllerClass, 'url_rule', Config::FIRST_SET);
+
+        if ($urlSegment) {
+            // Make director rule
+            if ($urlRule[0] == '/') $urlRule = substr($urlRule, 1);
```
Comment (kinglozzer) [inline admin/code/AdminRootController.php]: R.e. control structures:

> The body of each structure MUST be enclosed by braces.
http://www.php-fig.org/psr/psr-2/#5-control-structures

I’m guessing `php-cs-fixer` can’t do this automatically :disappointed:

File: admin/code/AdminRootController.php (hunk 1 above)
Comment (assertchris) [inline admin/code/AdminRootController.php]: My bad. I did the conversation by hand, and forgot that rule.

File: admin/code/AdminRootController.php (hunk 2 above)
Comment (assertchris) [inline admin/code/AdminRootController.php]: My bad. I did the conversation by hand, and forgot that rule.

File: admin/code/AdminRootController.php (hunk 1 above)
Comment (assertchris) [inline admin/code/AdminRootController.php]: Made all the non-breaking changes I could. I think we need to employ a better migration strategy than "convert everything and fix the breakages". :)

File: admin/code/AdminRootController.php (hunk 2 above)
Comment (dhensby) [inline admin/code/AdminRootController.php]: > I did the conversation by hand

Hardcore

**Pull request closed without merging**